internal/host/plugin/repository_host_catalog.go

```diff
@@ -59,23 +62,38 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
 	}
 	c.PublicId = id
 
-	oplogWrapper, err := r.kms.GetWrapper(ctx, c.ScopeId, kms.KeyPurposeOplog)
+	plgClient, ok := r.plugins[plg.GetPublicId()]
+	if !ok {
+		return nil, errors.New(ctx, errors.Internal, op, fmt.Sprintf("expected plugin %q not available", plg.GetPluginName()))
+	}
+	plgHc, err := toPluginCatalog(ctx, c)
 	if err != nil {
-		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get oplog wrapper"))
+		return nil, errors.Wrap(ctx, err, op)
 	}
-	dbWrapper, err := r.kms.GetWrapper(ctx, c.ScopeId, kms.KeyPurposeDatabase)
+	plgResp, err := plgClient.OnCreateCatalog(ctx, &plgpb.OnCreateCatalogRequest{Catalog: plgHc})
 	if err != nil {
-		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get db wrapper"))
+		return nil, errors.Wrap(ctx, err, op)
 	}
+	persistedData := plgResp.GetPersisted().GetData()
 
 	var hcSecret *HostCatalogSecret
-	if c.secrets != nil {
-		// TODO: Create the secret using the returned value from the call to the plugin.
-		hcSecret, err = newHostCatalogSecret(ctx, id, c.secrets)
+	if persistedData != nil {
+		hcSecret, err = newHostCatalogSecret(ctx, id, persistedData.AsMap())
```

Comment:
Rather than converting to a map and then json marshaling, we should proto marshal directly from the Struct.